invokeai/app/cli/commands.py

```diff
@@ -130,11 +91,12 @@ class ExitCli(Exception):
     pass
 
 
-class BaseCommand(ABC, BaseModel):
+class BaseCommand(ABC, InvokeAISettings):
     """A CLI command"""
 
     # All commands must include a type name like this:
     # type: Literal['your_command_name'] = 'your_command_name'
+    id: str = Field(description="The id of this command. Needed to allow the CLI to treat commands & nodes similarly.")
```

Comment:
CLI commands shouldn't have an id, as they aren't ever part of the graph, and aren't persisted after execution.

Reply:
Right. Will remove.

Reply:
By the way, `cli_app.py` tries to add the attributes `link` and `link_node` to invocations, but since these are not defined in the base invocation, this causes a validation failure. Should these be added? Previously the `allow_extra` Configuration field was set in the base class, but this can't be correct because it defeats the purpose of type checking.

Reply:
Hrm... `link` and `link_node` both worked for me last time I tried them. They're fairly complex, but effectively look at the node creation history to create edges to/from the node that's being created. I don't know why they'd have worked for me but don't work for you. Do they work on `main` for you?